enclave-runtime/src/lib.rs

```diff
@@ -759,32 +768,29 @@ where
 		)));
 	}
 
-	prepare_and_send_xts_and_block::<_, SignedSidechainBlock, _, _>(
-		validator,
-		on_chain_ocall_api,
-		calls,
-		Default::default(),
-		nonce,
-	)
+	prepare_and_send_xts::<_, _, _, Signer>(validator, signer, on_chain_ocall_api, calls, nonce)
 }
 
-fn prepare_and_send_xts_and_block<Block, SB, V, OCallApi>(
+fn prepare_and_send_xts<Block, V, OCallApi, Signer>(
 	validator: &mut V,
+	signer: Signer,
 	ocall_api: &OCallApi,
 	calls: Vec<OpaqueCall>,
-	blocks: Vec<SB>,
-	nonce: &mut u32,
-) -> Result<()>
+	nonce: u32,
+) -> Result<u32>
 where
 	Block: BlockT<Hash = H256>,
-	SB: SignedBlockT + 'static,
 	NumberFor<Block>: BlockNumberOps,
 	V: Validator<Block> + LightClientState<Block>,
 	OCallApi: EnclaveOnChainOCallApi,
+	Signer: Pair<Public = sp_core::ed25519::Public>,
+	Signer::Public: Encode,
+	Signer::Signature: Into<MultiSignature>,
 {
 	// store extrinsics in light client for finalization check
-	let extrinsics = create_extrinsics::<Block>(
+	let (extrinsics, nonce_updated) = create_extrinsics::<Block, Signer>(
```

Comment:
Just out of curiosity: But why is it better to have a return value instead of a mutable input parameter?

Reply:
So software design and clean code teaches that the result of a function should be the returned value, and parameters are for input only (see [here](https://dzone.com/articles/clean-code-robert-c-martins-way), in section `Functions`).
* Here we use `out` parameters as well as return values. This mixing is even harder to read and easy to miss as a caller.
* Having a mutable references to a parameter does not make it clear if it's an input AND output parameter, or just an output parameter. It is unclear who is responsible for allocating memory on that parameter.
* Returned value results allows the chaining of functions (as often used in functional paradigms, such as `map()`, `filter()` etc)

Reply:
I see, thank you for the detailed answer!